cryosparc2/protocols/protocol_cryosparc2d.py

```diff
@@ -423,8 +372,61 @@ def _updateClass(self, class2D):
             class2Drep.setLocation(index, fn)
             class2Drep.setSamplingRate(class2D.getSamplingRate())
 
+    def _createModelFile(self):
+        with open(self._getFileName('out_class'), 'r') as input_file, \
+                open(self._getFileName('out_class_m2'), 'w') as output_file:
+            j = 0  # mutex lock
+            i = 0  # start
+            for line in input_file:
+                if line.startswith("_rln"):
+                    output_file.write(line)
+                    i = 1
+                elif i == 0:
+                    output_file.write(line)
+                else:
+                    row = "%s %s@%s/%s %s\n"
+                    if j == 0:
+                        for n, m in enumerate(line.split()):
+                            if '@' in m:
+                                break
+                        j = 1
+                    output_file.write(row % (" ".join(line.split()[:n]),
```

Comment:
this function code looks horrifying, but I don't know maybe you have no other choice :)

Reply:
hehehehehe, I need to parse a plane file in order to create another file. I need to read line by line and split all of them, and finally compose a new line with the correct format.....However, I can take a look at it